src/actions.ts

```diff
@@ -16,58 +15,47 @@ export const openEnvironmentInBrowser = ({ id, projectId }: Environment) => {
   );
 };
 
-export const abortEnvironmentDeploy = (env: Environment) => {
-  const apiKeyCredentials = getApiKeyCredentials();
+export const abortEnvironmentDeploy = (
+  env: Environment,
+  apiClient: ApiClient
+) => {
   const id = env?.latestDeploymentLogId;
 
   if (!id) {
     return;
   }
-
-  const abortDeploymentUrl = `https://${ENV0_API_URL}/environments/deployments/${id}/abort`;
-  axios.post(abortDeploymentUrl, {}, { auth: apiKeyCredentials });
+  apiClient.abortDeployment(id);
 };
 
-export const cancelDeployment = (env: Environment) => {
-  const apiKeyCredentials = getApiKeyCredentials();
+export const cancelDeployment = (env: Environment, apiClient: ApiClient) => {
```

Comment:
@meniRoy how about creating a global API client and importing it? it reduces the boilerplates

Reply:
api client is dependent on `authService` and  `authService` needs `vscode.ExtensionContext`.
thats why i create the instance of `authService` and `apiClient`  in the activate function and pass them.
i can create the instance of `authService` and `apiClient`  directly in the file and export them and inject the dependencies in the activate function, WDYT?

Reply:
IMO the API client shouldn't know the auth service, it should instead get the auth credentials as an argument in the constructor ( optional ).

Reply:
ddb32f23aa4c469cc3080f29e0d16b74e2975abb